src/utils/validateTx.ts

```diff
@@ -1,19 +1,21 @@
-import { getTx } from '../web3/ether';
+import { getTx } from '../web3';
 import { TxData } from '../../types';
+import { ethers } from 'ethers';
 
 
-export function validateTx(txData:TxData):Promise<string>{
-    let res = getTx(txData.txHash)
-             .then((val:any)=>{
-                let value = val.map((v:any)=>{
-                    if(v) {
-                        return v;
-                    }
-                 })
-                 return value;
-             })
-             .catch((err:any)=>{
-                 return err;
-             });
-    return res;
+export async function validateTx({txHash,recipient,amount,invoiceId,network}:TxData):Promise<boolean> {
+    let res = await getTx(txHash,network);
+    let result = false;
+    let value = res.map((v:any)=>{
+        if(v && v.args.data) {   
+            let comparingEntity = JSON.parse(ethers.utils.toUtf8String(v.args.data))
+            let dataAmount = parseInt(v.args.amount._hex,16)
+            if(comparingEntity.invoiceId===invoiceId &&
```

Comment:
if it is not json it doesnt have the .invoiceId field

Reply:
The input is in form of string which has valid JSON I parsed the data and hence using the same to cross check invoiceID. If not this way then how should it get compared?

Reply:
when generating the payment link the invoiceid or data fields are expected to be plain strings, and on the wallet when making the transaction they are passed as plain strings to the transaction data field.
`ethers.utils.toUtf8String(v.args.data)===invoiceId`

Reply:
is the string a valid JSON?
I am using the same parser on the string to read the invoiceID from the stirng. @sirpy

Reply:
Do you mean that data field completely is our invoiceID?
I know the input example on data parameter initially shared was like 
```
{callbackUrl:'https://somevendor.com/api/mycallbackendpoint', invoiceId:'receipt-id-3938383',
wesite:'https://somevendor.com',
vendorName: 'Some Vendor',
}
```
Although this input is not a valid string for JSON parsing I created the hash from my end which had valid string for parsing.

Reply:
yes the data field in the tx will contain only the invoiceid

Reply:
Got it!
Thanks